pyinfra/api/connectors/dockerssh.py

```diff
@@ -0,0 +1,235 @@
+import os
+from tempfile import mkstemp
+
+import click
+import six
+
+from pyinfra import logger
+from pyinfra.api import QuoteString, StringCommand
+from pyinfra.api.exceptions import ConnectError, InventoryError, PyinfraError
+from pyinfra.api.util import get_file_io, memoize
+from pyinfra.progress import progress_spinner
+
+from . import ssh
+from .util import make_unix_command
+
+def remote_remove(state, host, filename, print_output=False, print_input=False):
+    remove_status, _, remove_stderr = run_shell_command(
+        state, host, 'rm -f {0}'.format(filename),
+        print_output=print_output,
+        print_input=print_input)
+
+    if not remove_status:
+        raise IOError('\n'.join(remove_stderr))
+
+def remote_mkstemp(state, host):
```

Comment:
We should avoid using Python on the remote side here, as this increases the remote requirements - I think just uploading to a temp filename works (see: https://github.com/Fizzadar/pyinfra/blob/master/pyinfra/api/connectors/ssh.py#L399).